### PR TITLE
refactor: code tweaks

### PR DESCRIPTION
**Pull request by HW13**
### Summary
- Modify `validateHMAC` to only accept an optional callback param.
- Add `validateRequests` options to enable preValidation hook.
- Remove `verificationErrorMessage` option.
- Modify `verificationError` to accept function that returns Error.
- Update README and examples.
- Add tests for `index.js`

### Test plan
- Confirm tests pass with 100% coverage
- Run the basic example:
```shell
npm run example:basic
```
- In another terminal session make the following request using `httpie`
```sh
http POST localhost:3000/foo Signature:'keyId="test-key-a", algorithm="hs2019", headers="(request-target) (created) (expires) host", signature="fKj7/7DbxLi1RkMqBrPB6HJ6dXUThRM4nwSErMo3ni1SuXZjTEpNDiZozUMUYqpcGssakLQl2y8ryBz0U5oxlg==", created=1402170695, expires=1402170895'
```
- The response should resemble the following:
```sh
HTTP/1.1 200 OK
Connection: keep-alive
Date: Tue, 15 Sep 2020 20:03:28 GMT
content-length: 13
content-type: application/json; charset=utf-8

{
    "foo": "bar"
}
```
- Try modifying the request signature and making another request
- The response should resemble the following:
```sh
HTTP/1.1 401 Unauthorized
Connection: keep-alive
Date: Tue, 15 Sep 2020 20:08:13 GMT
content-length: 83
content-type: application/json; charset=utf-8

{
    "error": "Unauthorized",
    "message": "Signature verification failed",
    "statusCode": 401
}
```

### CODE REVIEW

**Comment by eadmundo** [comment]
> * Remove `verificationErrorMessage` option.

how come?

File: README.md [hunk 1]
```diff
@@ -3,302 +3,95 @@
...
-      - A `getAlgorithm` method - [default](#getAlgorithm-method): A method that returns an algorithm string from the `algorithmMap` configuration object.
-      - A `getSignatureEncoding` method - [default](#getSignatureEncoding-method): A method that returns a `'base64'` encoding string to be used during HMAC signature construction.
-- Add a [global](#global-hook-example) or [route level](#route-level-hook-example) `preValidation` hook to your application.
-  - **Note:** For verification of HMAC signatures that include a body Digest header as HMAC key material, the `validateHMAC` step must take place on the `preValidation` lifecycle step as the fastify request body parsing takes place just prior to `preValidation`. Prior to `preValidation`, `request.body` will always be `null`.
+## Usage
 
-## Global Hook Example
+Registering the plugin, will decorate the `fastify` request with a `validateHMAC` method, and add a pre-validation hook to verify the HMAC signature of requests.
```
Comment (eadmundo) [inline README.md:20]:
```suggestion
Registering the plugin will decorate the `fastify` request instance with a `validateHMAC` method, and add a pre-validation hook to verify the HMAC signature of requests.
```

**Comment by HW13** [comment]
> > * Remove `verificationErrorMessage` option.
> 
> how come?

With the changes to the `verificationError` option I figured if a custom error message was desired, the user could just pass in `{ verificationError: () => new Error('Custom message') }`.

**Review by eadmundo — APPROVED** [review]
LGTM. Nice 🦐 ing

**Pull request merged**